### PR TITLE
Custom pos alibi flash attn fix

### LINKED ISSUE #298: [Bug] Custom alibi position fails for flash attention
Hi! I've been revisiting the custom position alibi again, and I found that setting `attn_flash=True` will fail. It looks like the following line forgot to account for the attention bias being 4D when custom positions are passed in. Should be an easy fix?
https://github.com/lucidrains/x-transformers/blob/57efd7770f2f5df0ff7b4ffcbd623750b584e850/x_transformers/attend.py#L373

### PR DESCRIPTION
**Pull request by Aceticia**
Fix #298

### CODE REVIEW

File: x_transformers/attend.py [hunk 1]
```diff
@@ -370,7 +370,7 @@ def flash_attn(
         # convert from bool to float
 
         if exists(attn_bias):
-            attn_bias = rearrange(attn_bias, 'h i j -> 1 h i j').expand(batch, heads, -1, -1)
+            attn_bias = attn_bias.expand(batch, heads, -1, -1)
```
Comment (lucidrains) [inline x_transformers/attend.py:373]: cool, but does this still work if the bias has 3 dimensions?

Comment (Aceticia) [reply x_transformers/attend.py:373]: yes, expand will broadcast along the last dimensions. I can add a more verbose rearrange back if you prefer that

Comment (lucidrains) [reply x_transformers/attend.py:373]: ah nice, didn't know that about expand! til

**Comment by lucidrains** [comment]
lgtm 🛥️

**Pull request merged**